### DISCUSSION TITLE
Issue with Backblaze B2 Connection Profile

### DISCUSSION THREAD

**Discussion by itswzyss** [iterate-ch/cyberduck#15743] [Q&A]
I've been pulling my hair out trying to get Cyberduck to recognize my B2 credentials. After generating new keys, I noticed Cyberduck was still not authorizing. After much closer inspection, I noticed that the default profile uses the following hostname: `s3.us-west-001.backblazeb2.com`. In my case, this is not correct for my buckets. I also noted that you cannot change this default value.

![Cyberduck_ihAxXkqm1w](https://github.com/iterate-ch/cyberduck/assets/25228751/90ab3e3e-0745-4162-b8d3-da8f320bd8cf)

Attempting to use this hostname results in this error:

![Cyberduck_LD8y0fsbWm](https://github.com/iterate-ch/cyberduck/assets/25228751/6ebf13c6-d564-4ed9-94b1-60158d628cd5)

After editing the connection profile located in %appdata%\Cyberduck\Profiles and allowing the `Hostname Configurable` field to be changed, I am then able to use the proper hostname which is `s3.us-west-002.backblazeb2.com`. Now my key works.

![Cyberduck_iU3mYnDYNa](https://github.com/iterate-ch/cyberduck/assets/25228751/e314ca78-fa9f-4265-818c-954f95e15d32)

Shouldn't the hostname just be left as a configurable option to avoid issues like this?

**Comment by dkocher** [comment]
This is referring to the S3 API compatible connection profile for Backblaze B2 [^1]. Curently the connection profile made available [^2]  is restricted to the US West region `s3.us-west-001.backblazeb2.com`. We should add support for other [account regions](https://www.backblaze.com/docs/cloud-storage-data-regions) for which I have opened iterate-ch/profiles#80 [^2].
 
[^1]: https://docs.cyberduck.io/protocols/b2/#using-s3-compatible-api
[^2]: iterate-ch/profiles#80
